src/braket/aws/aws_session.py

```diff
@@ -201,14 +207,11 @@ def add_braket_user_agent(self, user_agent: str) -> None:
         if user_agent not in self._braket_user_agents:
             self._braket_user_agents = f"{self._braket_user_agents} {user_agent}"
 
-    def _add_braket_user_agents_handler(self, request: awsrequest.AWSRequest, **kwargs) -> None:
-        try:
-            initial_user_agent = request.headers["User-Agent"]
-            request.headers.replace_header(
-                "User-Agent", f"{initial_user_agent} {self._braket_user_agents}"
-            )
-        except KeyError:
-            request.headers.add_header("User-Agent", self._braket_user_agents)
+        new_user_agent_config = Config(user_agent_extra=self._braket_user_agents)
```

Comment:
Where is this method actually used?

Reply:
The Qiskit and PL plugins use it:
- https://github.com/search?q=repo%3Aamazon-braket%2Famazon-braket-pennylane-plugin-python%20add_braket_user_agent&type=code
- https://github.com/search?q=repo%3Aqiskit-community%2Fqiskit-braket-provider%20add_braket_user_agent&type=code